### PR TITLE
MODBULKOPS-26: Update the module to Spring boot v3.0.0 and identify issues

### PR DESCRIPTION
**Pull request by MurodjonMukhamadaliev**
## Purpose
https://issues.folio.org/browse/MODBULKOPS-26: Update the module to Spring boot v3.0.0 and identify issues.

## Approach
Migrating to Spring Boot v3.0.0: https://wiki.folio.org/display/~Taras_Spashchenko/Migration+to+Spring+Boot+v3.0.0+How+to?src=contextnavpagetreemode

### TODOS and Open Questions

 - [ ] Use GitHub checklists. When solved, check the box and explain the answer.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/bulkops/adapters/ModClientAdapterFactory.java [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import java.util.List;
 import java.util.Map;
 
-import javax.annotation.PostConstruct;
+import jakarta.annotation.PostConstruct;
```
Comment (tatsiana-tarhonskaya) [inline src/main/java/org/folio/bulkops/adapters/ModClientAdapterFactory.java:6]: please double check imports in all files. It seems to me that existing alphabetical order is broken

File: pom.xml [hunk 2]
```diff
@@ -33,7 +33,7 @@
     <argLine/>
     <bulk-operations.yaml.file>${project.basedir}/src/main/resources/swagger.api/bulk-operations.yaml</bulk-operations.yaml.file>
 
-    <folio-spring-base.version>5.0.2</folio-spring-base.version>
+    <folio-spring-base.version>6.0.0</folio-spring-base.version>
     <jackson-bom.version>2.14.1</jackson-bom.version>
```
Comment (tatsiana-tarhonskaya) [inline pom.xml:37]: did you check other dependenies' updates? Looks like at least spring-kafka.version should have third version released

File: src/test/resources/application.yml [hunk 3]
```diff
@@ -35,3 +39,4 @@ application:
     group-id: ${ENV:folio}-mod-bulk-operations-events-group
   file-uploading:
     max-retry-count: ${MAX_UPLOADING_RETRIES:50}
+
```
Comment (tatsiana-tarhonskaya) [inline src/test/resources/application.yml:42]: please remove the line

File: src/main/java/org/folio/bulkops/adapters/ModClientAdapterFactory.java (hunk 1 above)
Comment (MurodjonMukhamadaliev) [reply src/main/java/org/folio/bulkops/adapters/ModClientAdapterFactory.java:6]: I haven't changed order in almost all files, just changed javax => jakarta

**Review by TarasSpashchenko — APPROVED** [review]

**Review by khandramai — APPROVED** [review]

**Pull request merged**